From e01b59c7accd6e6984d16b8ed836b70001ebf6e1 Mon Sep 17 00:00:00 2001
From: "Lu.Jiang" <lu.jiang@windriver.com>
Date: Fri, 12 Sep 2014 14:12:39 +0800
Subject: [PATCH 548/987] dpa_offload: Sync B4420QDS offloading DTS files

[Original patch taken from QorIQ-SDK-V1.6-SOURCE-20140619-yocto.iso]

Updated the B4420QDS DTS example files for DPA offloading applications
to bring them to the version delivered with SDK 1.5.

Signed-off-by: Marian Chereji <marian.chereji@freescale.com>
Change-Id: Ic40b3f4f9d5e4df9213918ce7aed2dca99423465
Reviewed-on: http://git.am.freescale.net:8181/8836
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Radu-Andrei Bulie <Radu.Bulie@freescale.com>
Reviewed-by: Jose Rivera <German.Rivera@freescale.com>
Signed-off-by: Lu.Jiang <lu.jiang@windriver.com>
---
 .../dts/b4420qds-usdpaa-shared-interfaces.dts      | 52 ++++++++++++----------
 .../fsl_dpa_offload/dts/b4420qds-usdpaa.dts        |  5 +--
 .../fsl_dpa_offload/dts/b4420si-chosen-offld.dtsi  |  2 +
 .../fsl_dpa_offload/dts/b4420si-chosen-reass.dtsi  |  2 +
 .../staging/fsl_dpa_offload/dts/b4420si-pre.dtsi   | 23 ++++++++++
 5 files changed, 57 insertions(+), 27 deletions(-)

diff --git a/drivers/staging/fsl_dpa_offload/dts/b4420qds-usdpaa-shared-interfaces.dts b/drivers/staging/fsl_dpa_offload/dts/b4420qds-usdpaa-shared-interfaces.dts
index 1361777..e19691e 100644
--- a/drivers/staging/fsl_dpa_offload/dts/b4420qds-usdpaa-shared-interfaces.dts
+++ b/drivers/staging/fsl_dpa_offload/dts/b4420qds-usdpaa-shared-interfaces.dts
@@ -77,25 +77,23 @@
 		fsl,bpid = <12>;
 		fsl,bpool-thresholds = <0x100 0x300 0x0 0x0>;
 	};
-
 	bp16: buffer-pool@16 {
-                        compatible = "fsl,b4420-bpool", "fsl,bpool";
-                        fsl,bpid = <16>;
-			fsl,bpool-ethernet-cfg = <0 2048 0 1728 0 0>;
-                        fsl,bpool-thresholds = <0x100 0x300 0x0 0x0>;
-       };
-
+		compatible = "fsl,b4420-bpool", "fsl,bpool";
+		fsl,bpid = <16>;
+		fsl,bpool-ethernet-cfg = <0 2048 0 1728 0 0>;
+		fsl,bpool-thresholds = <0x100 0x300 0x0 0x0>;
+	};
 
 	fsl,dpaa {
 		ethernet@0 {
 			compatible = "fsl,b4420-dpa-ethernet-init", "fsl,dpa-ethernet-init";
-			fsl,bman-buffer-pools = <&bp7 &bp8 &bp9>;
+			fsl,bman-buffer-pools = <&bp16>;
 			fsl,qman-frame-queues-rx = <0x50 1 0x51 1>;
 			fsl,qman-frame-queues-tx = <0x70 1 0x71 1>;
 		};
 		ethernet@1 {
 			compatible = "fsl,b4420-dpa-ethernet-init", "fsl,dpa-ethernet-init";
-			fsl,bman-buffer-pools = <&bp7 &bp8 &bp9>;
+			fsl,bman-buffer-pools = <&bp16>;
 			fsl,qman-frame-queues-rx = <0x52 1 0x53 1>;
 			fsl,qman-frame-queues-tx = <0x72 1 0x73 1>;
 		};
@@ -118,20 +116,26 @@
 		 * On receiving frame at this interface, the ethernet driver will do
 		 * kmap_atomic/kunmap_atomic for that frame. */
 		ethernet@16 {
-			compatible = "fsl,b4420-dpa-ethernet", "fsl,dpa-ethernet";
+			compatible = "fsl,b4420-dpa-ethernet-macless", "fsl,dpa-ethernet-macless";
 			fsl,bman-buffer-pools = <&bp16>;
 			fsl,qman-frame-queues-rx = <4000 8>;
 			fsl,qman-frame-queues-tx = <4008 8>;
 			local-mac-address = [00 11 22 33 44 55];
 		};
 		ethernet@17 {
-			compatible = "fsl,b4420-dpa-ethernet", "fsl,dpa-ethernet";
+			compatible = "fsl,b4420-dpa-ethernet-macless", "fsl,dpa-ethernet-macless";
 			fsl,bman-buffer-pools = <&bp16>;
 			fsl,qman-frame-queues-rx = <5000 8>;
 			fsl,qman-frame-queues-tx = <5008 8>;
 			local-mac-address = [00 11 22 33 44 66];
 		};
-
+		ethernet@18 {
+			compatible = "fsl,b4420-dpa-ethernet-macless", "fsl,dpa-ethernet-macless";
+			fsl,bman-buffer-pools = <&bp16>;
+			fsl,qman-frame-queues-rx = <6000 8>;
+			fsl,qman-frame-queues-tx = <6008 8>;
+			local-mac-address = [00 11 22 33 44 77];
+		};
 		dpa-fman0-oh@2 {
 			compatible = "fsl,dpa-oh";
 			/* Define frame queues for the OH port*/
@@ -140,17 +144,17 @@
 			fsl,bman-buffer-pools = <&bp16>;
 			fsl,fman-oh-port = <&fman0_oh2>;
 		};
-                dpa_fman0_oh3: dpa-fman0-oh@3 {
-                        compatible = "fsl,dpa-oh";
-                        fsl,qman-frame-queues-oh = <0x68 1 0x69 1>;
-                        fsl,bman-buffer-pools = <&bp9>;
-                        fsl,fman-oh-port = <&fman0_oh3>;
-                };
-                dpa_fman0_oh4: dpa-fman0-oh@4 {
-                        compatible = "fsl,dpa-oh";
-                        fsl,qman-frame-queues-oh = <0x70 1 0x71 1>;
-                        fsl,bman-buffer-pools = <&bp9>;
-                        fsl,fman-oh-port = <&fman0_oh4>;
-                };
+		dpa_fman0_oh3: dpa-fman0-oh@3 {
+			compatible = "fsl,dpa-oh";
+			fsl,qman-frame-queues-oh = <0x68 1 0x69 1>;
+			fsl,bman-buffer-pools = <&bp9>;
+			fsl,fman-oh-port = <&fman0_oh3>;
+		};
+		dpa_fman0_oh4: dpa-fman0-oh@4 {
+			compatible = "fsl,dpa-oh";
+			fsl,qman-frame-queues-oh = <0x70 1 0x71 1>;
+			fsl,bman-buffer-pools = <&bp9>;
+			fsl,fman-oh-port = <&fman0_oh4>;
+		};
 	};
 };
diff --git a/drivers/staging/fsl_dpa_offload/dts/b4420qds-usdpaa.dts b/drivers/staging/fsl_dpa_offload/dts/b4420qds-usdpaa.dts
index 0b7b71b..8165f70 100644
--- a/drivers/staging/fsl_dpa_offload/dts/b4420qds-usdpaa.dts
+++ b/drivers/staging/fsl_dpa_offload/dts/b4420qds-usdpaa.dts
@@ -77,7 +77,6 @@
 		fsl,bpid = <12>;
 		fsl,bpool-thresholds = <0x100 0x300 0x0 0x0>;
 	};
-
 	bp16: buffer-pool@16 {
 		compatible = "fsl,b4420-bpool", "fsl,bpool";
 		fsl,bpid = <16>;
@@ -117,14 +116,14 @@
 		 * On receiving frame at this interface, the ethernet driver will do
 		 * kmap_atomic/kunmap_atomic for that frame. */
 		ethernet@16 {
-			compatible = "fsl,b4420-dpa-ethernet", "fsl,dpa-ethernet";
+			compatible = "fsl,b4420-dpa-ethernet-macless", "fsl,dpa-ethernet-macless";
 			fsl,bman-buffer-pools = <&bp16>;
 			fsl,qman-frame-queues-rx = <4000 8>;
 			fsl,qman-frame-queues-tx = <4008 8>;
 			local-mac-address = [00 11 22 33 44 55];
 		};
 		ethernet@17 {
-			compatible = "fsl,b4420-dpa-ethernet", "fsl,dpa-ethernet";
+			compatible = "fsl,b4420-dpa-ethernet-macless", "fsl,dpa-ethernet-macless";
 			fsl,bman-buffer-pools = <&bp16>;
 			fsl,qman-frame-queues-rx = <5000 8>;
 			fsl,qman-frame-queues-tx = <5008 8>;
diff --git a/drivers/staging/fsl_dpa_offload/dts/b4420si-chosen-offld.dtsi b/drivers/staging/fsl_dpa_offload/dts/b4420si-chosen-offld.dtsi
index b47d8fb..91ec766 100644
--- a/drivers/staging/fsl_dpa_offload/dts/b4420si-chosen-offld.dtsi
+++ b/drivers/staging/fsl_dpa_offload/dts/b4420si-chosen-offld.dtsi
@@ -10,6 +10,7 @@ chosen {
 			fman0_oh2-extd-args {
 				cell-index = <1>;
 				compatible = "fsl,fman-port-op-extended-args";
+				errors-to-discard = <0x00420000>;
 				/* Define Virtual storage profile */
 				/* <number of profiles, default profile id> */
 				vsp-window = <8 0>;
@@ -18,6 +19,7 @@ chosen {
 			fman0_oh3-extd-args {
 				cell-index = <2>;
 				compatible = "fsl,fman-port-op-extended-args";
+				errors-to-discard = <0x00420000>;
 				/* Define buffer layout parameters. Can be used
 				 * in fragmentation or header manip operations
 				 */
diff --git a/drivers/staging/fsl_dpa_offload/dts/b4420si-chosen-reass.dtsi b/drivers/staging/fsl_dpa_offload/dts/b4420si-chosen-reass.dtsi
index 9d65380..81b8b9b 100644
--- a/drivers/staging/fsl_dpa_offload/dts/b4420si-chosen-reass.dtsi
+++ b/drivers/staging/fsl_dpa_offload/dts/b4420si-chosen-reass.dtsi
@@ -9,6 +9,7 @@ chosen {
 			fman0_rx3-extd-args {
 				cell-index = <3>;
 				compatible = "fsl,fman-port-1g-rx-extended-args";
+				errors-to-discard = <0x00420000>;
 				/* Define Virtual storage profile */
 				/* <number of profiles, default profile id> */
 				vsp-window = <8 0>;
@@ -16,6 +17,7 @@ chosen {
 			fman0_oh2-extd-args {
 				cell-index = <1>;
 				compatible = "fsl,fman-port-op-extended-args";
+				errors-to-discard = <0x00420000>;
 				/* Define buffer layout parameters. Can be used
 				 * in fragmentation or header manip operations
 				 */
diff --git a/drivers/staging/fsl_dpa_offload/dts/b4420si-pre.dtsi b/drivers/staging/fsl_dpa_offload/dts/b4420si-pre.dtsi
index cc7cf06..52c7613 100644
--- a/drivers/staging/fsl_dpa_offload/dts/b4420si-pre.dtsi
+++ b/drivers/staging/fsl_dpa_offload/dts/b4420si-pre.dtsi
@@ -90,4 +90,27 @@
 			next-level-cache = <&L2>;
 		};
 	};
+
+	dsp-clusters {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		dsp-cluster0 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			compatible = "fsl,sc3900-cluster";
+			reg = <0>;
+
+			dsp0: dsp@0 {
+				compatible = "fsl,sc3900";
+				reg = <0>;
+				next-level-cache = <&L2_2>;
+			};
+			dsp1: dsp@1 {
+				compatible = "fsl,sc3900";
+				reg = <1>;
+				next-level-cache = <&L2_2>;
+			};
+		};
+	};
 };
-- 
1.9.1

